doc: enhance cross compile section in getting started

yashi opened the pull request:
"Getting Started" does not clearly state how to use a custom cross
compiler.  This patch adds a new section "Using Custom Cross
Compilers" to it.

To support users without SDK, we have the section "Building without
the Zephyr SDK".  But the section is all about prebuilt tools in the
SDK but not build process.  This patch adds two pointers "Using 3rd
Party Cross Compilers" and "Using Custom Cross Compilers" pointing to
the actual building process, and let those section describe
ZEPHYR_TOOLCHAIN_VARIANT, which is cross compilers dependent variable.

The patch also clarify what a 3rd party cross compiler is and what a
custom compiler is.

TODO: I don't know what custom compilers are already working on MacOS
and Windows.  It'd be nice to have a working example on each platform.

TODO: On Fedora/Redhat, should I use "yum" or "dnf" for installation?
Plus, should I use "arm-none-eabi-newlib" or packages from
Codesourcery? https://fedoraproject.org/wiki/Embedded

SebastianBoe commented on doc/getting_started/getting_started.rst:
```diff
@@ -151,44 +151,47 @@ Building without the Zephyr SDK
+It is, however, possible to build the Zephyr OS without the SDK.  If you are
+using 3rd party crosss compilers, jump forward to `Using 3rd Party Cross
+Compilers`_ for details.  A "3rd party cross compiler" is a toolchain that the
+Zephyr build system already know, such as `GCC ARM Embedded`_ we use in this
```
'know'

knows about

yashi replied:
Not yet.  I'll update once I merged your suggestions.

SebastianBoe commented on doc/getting_started/getting_started.rst:
```diff
@@ -151,44 +151,47 @@ Building without the Zephyr SDK
+any extra flags when building applications or running tests. In addition to
+cross-compilers, the Zephyr SDK also provides prebuilt host tools.
+
+It is, however, possible to build the Zephyr OS without the SDK.  If you are
```
Not sure if we say Zephyr OS.

Maybe just 

possible to build without the Zephyr SDK.

It's implied that we are trying to build zephyr.

SebastianBoe commented on doc/getting_started/getting_started.rst:
```diff
@@ -151,44 +151,47 @@ Building without the Zephyr SDK
+using 3rd party crosss compilers, jump forward to `Using 3rd Party Cross
+Compilers`_ for details.  A "3rd party cross compiler" is a toolchain that the
+Zephyr build system already know, such as `GCC ARM Embedded`_ we use in this
+document.
```
Something is off with the english in 

 such as `GCC ARM Embedded`_ we use in this document.

Not sure how to improve.

yashi replied:
hmm..  would you mind to elaborate?  Does "such as GCC ARM Embedded *that* we use in this document." make sense to you?

SebastianBoe replied:
Yes, that's better.

yashi replied:
OK. fixed.

SebastianBoe commented on doc/getting_started/getting_started.rst:
```diff
@@ -211,6 +214,65 @@ without the Zephyr SDK`_ for details.
+      # On Windows
+      set GCCARMEMB_TOOLCHAIN_PATH=
+      set ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
+      set CROSS_COMPILE=/usr/bin/arm-none-eabi-
```
I think you can omit the Windows example.

/usr/bin/arm-none-eabi-

isn't a valid windows path.

SebastianBoe replied:
"Or, would it be OK to delete about windows from the custom compiler section?"
Yes

SebastianBoe reviewed (dismissed):
Review feedback is in comments.

yashi commented:
Thank you @SebastianBoe .  I've added my comment, too.

I commented above but who should I talk to about Windows support?  I don't have windows ATM and don't know cross compiler for it.  But I assume that Zephyr offers first class support for Windows.  IOW, it's not an option to remove console example for windows from the section, right?

SebastianBoe commented:
"I commented above but who should I talk to about Windows support?"

The procedure is the same for Windows users, so they should be able to figure this out based on the linux-specific docs.

SebastianBoe commented:
It is OK to remove:

```
# On Windows
set GCCARMEMB_TOOLCHAIN_PATH=
set ZEPHYR_TOOLCHAIN_VARIANT=cross-compile
set CROSS_COMPILE=/usr/bin/arm-none-eabi-
```

It is not OK to remove:

```
.. zephyr-app-commands::
   :zephyr-app: samples/hello_world
   :board: arduino_zero
   :goals: build

```

dbkinder reviewed (changes requested):
There's also a problem with the signature and Signed-off-by in your commit message.

yashi commented on doc/getting_started/getting_started.rst:
```diff
@@ -211,6 +214,65 @@ without the Zephyr SDK`_ for details.
+      :board: arduino_zero
+      :goals: build
+
+Note that the Zephyr bulid system assumes that your toolchain, a set
```
fixed.  and spell checked all again.  thanks.

yashi commented on doc/getting_started/getting_started.rst:
```diff
@@ -211,6 +214,65 @@ without the Zephyr SDK`_ for details.
+      :goals: build
+
+Note that the Zephyr bulid system assumes that your toolchain, a set
+of tools to compile and link your code, resides in a same directory
```
done

yashi commented on doc/getting_started/getting_started.rst:
```diff
@@ -211,6 +214,65 @@ without the Zephyr SDK`_ for details.
+
+Make sure to unset the ZEPHYR_SDK_INSTALL_DIR if you don't use the
+SDK's host tools. See `Building without the Zephyr SDK`_ for details.
+If you do, keep the ZEPHYR_SDK_INSTALL_DIR environment variable set to
```
I changed it to
```
Make sure to unset the ZEPHYR_SDK_INSTALL_DIR if you don't use the SDK's host
tools. See `Building without the Zephyr SDK`_ and `Set Up the Development
Environment`_ for more details.
```

yashi committed:
doc: enhance cross compile section in getting started

"Getting Started" does not clearly state how to use a custom cross
compiler.  This patch adds a new section "Using Custom Cross
Compilers" to it.

To support users without SDK, we have the section "Building without
the Zephyr SDK".  But the section is all about prebuilt tools in the
SDK but not build process.  This patch adds two pointers "Using 3rd
Party Cross Compilers" and "Using Custom Cross Compilers" pointing to
the actual building process, and let those section describe
ZEPHYR_TOOLCHAIN_VARIANT, which is cross compilers dependent variable.

The patch also clarify what a 3rd party cross compiler is and what a
custom compiler is.

Signed-off-by: Yasushi SHOJI <y-shoji@ispace-inc.com>

yashi commented:
merged all comments so far.

yashi commented:
@dbkinder anything else  I can do?

dbkinder reviewed (approved):
LGTM, thanks.

yashi commented:
removing the "DNM" tag on the title hoping someone will merge this
